src/besselj.jl

```diff
@@ -160,72 +88,26 @@ function besselj0(x::Float32)
-         (0.0, 0.18006337534431555, -0.004589739858905968, -0.029776581472313136, 0.0007530284838491313, 0.0014660390526475475, -3.658900932751018e-5, -3.4137236390211275e-5, 8.366325091910564e-7, 4.6108475352995054e-7, -1.1047200388967438e-8, -4.058793425595929e-9, 9.363513647344476e-11, 2.4720051360744732e-11),
-         (0.1734590492857464, -8.319859584484218e-17, -0.08653590193876039, 0.0013568189856187354, 0.007147216520714834, -0.00013295776598760658, -0.00023462959289705953, 4.6031437800947915e-6, 4.103196271673978e-6, -8.18373301294018e-8, -4.44403211204062e-8, 8.872617322304839e-10, 3.2359730989751983e-10, -7.354373969368619e-12),
-         (0.0, -0.16718460047381806, 0.0036727588017285762, 0.02770273166133465, -0.0006050364924617241, -0.0013693112504440895, 2.9615596890234806e-5, 3.206096184277923e-5, -6.841058767433935e-7, -4.358877331737043e-7, 9.143763763164119e-9, 3.864037532317281e-9, -7.851667555762691e-11, -2.3728757802646805e-11),
-         (-0.1618382095526585, 7.115722920799028e-17, 0.08078219522574913, -0.0011038527651152986, -0.006686444726790514, 0.00010870535083026479, 0.00022030197347228358, -3.791805814546456e-6, -3.871147931227952e-6, 6.80607764766372e-8, 4.2160129307780366e-8, -7.460427055677985e-10, -3.0872139357605097e-10, 6.24158293622982e-12),
-         (0.0, 0.15672498625285222, -0.0030251499811055965, -0.026004046442225856, 0.0004996832448037306, 0.001288697907658241, -2.457760943647847e-5, -3.02856199504406e-5, 5.715983781675481e-7, 4.136275513873783e-7, -7.703846567878601e-9, -3.6853569596797437e-9, 6.676439988396374e-11, 2.277101500922319e-11))#for i in 2:2:18
-
-J1_POLY_PIO2(::Type{Float64}) = (0.5, -0.0624999999999989, 0.002604166666657291, -5.42534721917933e-5, 6.781683542660179e-7, -5.651361336587487e-9, 3.36191211106159e-11, -1.4511302591871352e-13)
-
-#for i in 1:2:18
-#    display(((i+1)*pi/2-pts[i+1][1], (i+2)*pi/2-pts[i+1][1]))
-#    poly = Float64.(Tuple(ratfn_minimax(x->r2(x,pts[i]), (i*pi/2-pts[i][1], (i+1)*pi/2-pts[i][1]), 13, 0)[1]))
-#    push!(polys, poly)
-#    poly = Float64.(Tuple(ratfn_minimax(x->r(x,pts[i+1]), ((i+1)*pi/2-pts[i+1][1], (i+2)*pi/2-pts[i+1][1]), 12, 0)[1]))
-#    push!(polys, (0.0, poly...))
-#end
-
 """
     besselj1(x::T) where T <: Union{Float32, Float64}
 
 Bessel function of the first kind of order one, ``J_1(x)``.
 """
 function besselj1(x::Float64)
     T = Float64
+    s = sign(x)
     x = abs(x)
-    isinf(x) && return zero(x)
 
     if x <= 26.0
-        x <= pi/2 && return x*evalpoly(x*x, J1_POLY_PIO2(T))
-        n = unsafe_trunc(Int, TWOOPI(T)*x)
+        x <= pi/2 && return x * evalpoly(x * x, J1_POLY_PIO2(T))
+        n = unsafe_trunc(Int, TWOOPI(T) * x)
         root = @inbounds J1_ROOTS(T)[n]
         r = x - root[1] - root[2]
-        return evalpoly(r, @inbounds J1_POLYS(T)[n])
+        return evalpoly(r, @inbounds J1_POLYS(T)[n]) * s
     else
         xinv = inv(x)
-        x2 = xinv*xinv
+        iszero(xinv) && return zero(T)
```

Comment:
technically this should probably return `xinv` so it returns `-0.0` when applicable.

Reply:
I believe since we have `x=abs(x)` this would still always return `0.0` so might need to handle that separately. I'm actually starting to not appreciate that syntax of overwriting that variable and prefering `xabs = abs(x)`. Thoughts?

Reply:
I don't think it really matters. I would probably just merge this as is.